.travis.yml

```diff
@@ -10,44 +10,122 @@ env:
+      if: type != cron
       env: CI_METRICS=1 PLATFORM_VERSION=10.3 TEST=functional/web
 
     # smoke test on xcode 9.3 web
     - osx_image: xcode9.2
+      if: type != cron
       env: CI_METRICS=1 PLATFORM_VERSION=11.2 TEST=functional/web
+
+    # Sauce Emusim tests
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=0 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=0 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=0 TEST=functional/long
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=1 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=1 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=1 TEST=functional/long
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=2 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=2 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=2 TEST=functional/long
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=3 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=3 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=3 TEST=functional/long
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=4 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=4 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_EMUSIM_DEVICE_INDEX=4 TEST=functional/long
+
+    # Sauce RDC Tests
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=0 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=0 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=0 TEST=functional/long
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=1 TEST=functional/basic
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=1 TEST=functional/web
+    - os: linux
+      if: type = cron
+      env: CLOUD=1 SAUCE_RDC_DEVICE_INDEX=1 TEST=functional/long
+    
 git:
   submodules: false
 before_install:
   # Use sed to replace the SSH URL with the public URL, then initialize submodules
   # code from http://stackoverflow.com/a/24600210/375688
-  - sed -i '' 's/git@github.com:/https:\/\/github.com\//' /Users/travis/build/appium/appium-xcuitest-driver/.gitmodules
-  - git submodule update --init --recursive
+  - if [ "$TRAVIS_OS_NAME" != "linux" ]; then
```

Comment:
This check probably makes more sense to be based on `CLOUD` not on the OS, which we could want to change at some point.

Reply:
I changed it to: ```if [ "$TRAVIS_OS_NAME" = "osx" ]```

The reason for it is that we can't install WDA on non-OSX.

But if we are using OSX and Cloud simultaneously, we may as well still run that script for testing purposes.